source/PlayerInfo.cpp

```diff
@@ -1042,10 +1052,27 @@ void PlayerInfo::BuyShip(const Ship *model, const string &name, bool isGift)
 			for(const auto &it : model->Outfits())
 				stock[it.first] -= it.second;
 		}
+
+		DiscoverOutfits(model->Outfits());
 	}
 }
 
+void PlayerInfo::DiscoverOutfits(const std::map<const Outfit *, int> &outfits)
+{
+		for(const auto &it : outfits)
+			if(!OutfitIsKnown(*it.first))
+				DiscoverOutfit(*it.first);
+}
 
+void PlayerInfo::VisitOutfitter(const Sale<Outfit>& outfitter)
```

Comment:
convention is 3 empty lines between functions, you only have 1 here, above and below

Reply:
will fix